cylc/sphinx_ext/cylc_lang/autodocumenters.py

```diff
@@ -346,3 +367,314 @@ def run(self):
+            )
+        else:
+            sub = run(['cylc', 'config', '--json'], capture_output=True)
+
+        CylcMetadataDirective.check_subproc_output(sub)
+
+        return json.loads(sub.stdout)
+
+    @staticmethod
+    def get_global_metadata(config):
+        """
+        Additional Processing:
+        * Get lists of hosts/platforms and job runner from the config.
+        * If no title is provided, use the platform/group regex as the title.
+        * If title != regex then insert text saying which regex
+          needs matching to select this platform.
+
+        Returns:
+            A dictionary in the form:
+            'platforms': {'platform regex': {..metadata..}},
+            'platform groups': {'platform regex': {..metadata..}}
+        """
+        metadata = {}
+        for section, select_from in zip(
+            ['platforms', 'platform groups'],
+            ['hosts', 'platforms']
+        ):
+            metadata[section] = config.get(section)
+            if not metadata[section]:
+                continue
+            for key in config.get(section).keys():
+                # Grab a list of hosts or platforms that this
+                # platform or group will select from:
+                select_from = (
+                    config.get(section).get(key).get('hosts')
+                    or config.get(section).get(key).get('platforms'))
+                select_from = select_from or [key]
+                metadata[section][key]['select_from'] = select_from
+
+                # Grab the job runner this platform uses:
+                if section == 'platforms':
+                    metadata[section][key]['job_runner'] = config.get(
+                        section).get(key).get('job runner', 'background')
+        return metadata
+
+    @staticmethod
+    def global_to_node(meta, src):
+        """Convert the global metadata into rst format.
+        """
+        ret = [
+            f'.. cylc:conf:: Global Config: {src}', '',
+            f'{INDENT}.. note::', '',
+            f'{INDENT * 2}Platforms and platform groups are listed'
+            ' in the order in which Cylc will check for matches to the'
+            ' ``[runtime][NAMESPACE]platform`` setting.', '']
+
+        # For platforms in platform groupts then for hosts in platforms:
+        for settings, selects in zip(
+            ['platform groups', 'platforms'],
+            ['platforms', 'hosts']
+        ):
+            if meta.get(settings, {}):
+                ret += [f'{INDENT}.. cylc:conf:: {settings}', '']
+                for regex, info in reversed(meta[settings].items()):
+                    platform_meta = info.get('meta', {})
+                    title = platform_meta.get('title', '')
+                    if not title:
+                        title = regex
+                    ret += [
+                        f'{INDENT * 3}.. cylc:conf:: {title}', '',
+                        f'{INDENT * 4}.. csv-table:: Key Details', '']
```

Comment:
There's a native way to do this in ReST without using a directive:

```
:Key1: Value 1
:Key2: Value 2
```

For an example of this in motion, see the interventions section:

https://github.com/cylc/cylc-doc/blob/9e5c8f093e79b85b7e0d73c42ce55241f73ef110/src/user-guide/interventions/index.rst?plain=1#L94-L100

The title "Key Details" is probably unnecessary.